rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -417,7 +417,12 @@ public J visitMethodInvocation(J.MethodInvocation method, Integer integer) {
                     }
                     return m;
                 }
-                return maybeReplaceStatement(method, J.class, true, 0);
+                Cursor varDeclsCursor = getCursor().dropParentUntil(v -> v instanceof J.CompilationUnit
```

Comment:
Should check for `JavaSourceFile` instead of `CompilationUnit`

Reply:
For compatibility with Groovy source files (and eventually kotlin) that don't have a J.CompilationUnit at their root

Reply:
good catch thanks!